bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/MagnitudeConstraints.scala

```diff
@@ -45,36 +47,74 @@ object SaturationConstraint {
     Order.orderBy(_.brightness)
 }
 
+sealed trait MagnitudeFilter {
+  def filter(t: SiderealTarget): Boolean
+
+  def searchBands: BandsList
+  def faintnessConstraint: FaintnessConstraint
+  def saturationConstraint: Option[SaturationConstraint]
+}
+
 /**
- * Describes constraints for the magnitude of a target
- * See OT-19.
+ * Defines a typeclass of classes that can adjust MagnitudeConstraints, e.g. given certain conditions
  */
-case class MagnitudeConstraints(band: MagnitudeBand, faintnessConstraint: FaintnessConstraint, saturationConstraint: Option[SaturationConstraint]) {
+trait ConstraintsAdjuster[T] {
+  def adjust(t: T, mc: MagnitudeConstraints): MagnitudeConstraints
+}
 
-  def this(faint:Magnitude) = this(faint.band, FaintnessConstraint(faint.value), None)
+/**
+ * Defines a set of bands that can extract a magnitude from a target
+ */
+sealed trait BandsList {
+  val bands: NonEmptyList[MagnitudeBand]
+  def extract(t: SiderealTarget) = bands.map(t.magnitudeIn).list.find(_.isDefined).flatten
+  def bandSupported(b: MagnitudeBand) = bands.list.contains(b)
+}
 
-  /**
-   * Maps a transformation into a new MagnitudeConstraints
-   */
-  def map(f: Magnitude => Magnitude): MagnitudeConstraints = {
-    val mappedFaintness = f(faintnessConstraint.toMagnitude(band)) 
-    val mappedSaturation = saturationConstraint.map(_.toMagnitude(band)).map(f)
+/**
+ * Extracts the first valid R Band Magnitude if available
+ */
+case object RBandsList extends BandsList {
+  val bands = NonEmptyList(MagnitudeBand._r, MagnitudeBand.R, MagnitudeBand.UC)
+}
 
-    val fl = FaintnessConstraint(mappedFaintness.value)
-    val sl = mappedSaturation.map(s => SaturationConstraint(s.value))
+/**
+ * Extractor for Nici containing 4 bands
+ */
+case object NiciBandsList extends BandsList {
+  val bands = NonEmptyList(MagnitudeBand._r, MagnitudeBand.R, MagnitudeBand.UC, MagnitudeBand.K)
```

Comment:
Probably doesn't matter since NICI is retired but a question for Andy is whether R and K are really interchangeable.